pkg/global/apply_configuration.go

```diff
@@ -43,43 +47,94 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-// LifecycleState is returned by ApplyConfiguration. It can be used by
+const (
+	stateNotServing int32 = iota
+	stateServing
+)
+
+// ExitCodeInterrupted is used to signal a successful controlled shutdown.
+var ExitCodeInterrupted int = 8
```

Comment:
My preference would be to remove the signal handler and raise SIGTERM again. That way the calling process gets accurate process shutdown information.